lib/Core/BuildEngine.cpp

```diff
@@ -816,9 +816,9 @@ class BuildEngineImpl {
           std::string error;
           bool result = db->setRuleResult(ruleInfo->rule, ruleInfo->result, &error);
           if (!result) {
-            // FIXME: Error handling.
-            std::cerr << error << std::endl;
-            exit(1);
+            delegate.error(error);
+            completeRemainingTasks();
+            return false;
```

Comment:
When this happens, do you think we should try and take care to avoid committing the database transition? We could record that we aborted due to a db failure and not attempt a commit (which is likely to either be wrong, or fail).